client/src/pages/lab_management/announcements/AnnouncementModalContents.tsx

```diff
@@ -32,44 +31,73 @@ export default function AnnouncementModalContents({
   onDelete,
   loading,
 }: AnnouncementPageProps) {
-
   const navigate = useNavigate();
-  
+
+  const [showLinkFields, setShowLinkFields] = useState(!!announcementDraft.linkUrl || !!announcementDraft.linkText);
```

Comment:
When testing this, it seems like the default value of this isnt getting calculated just right. If i add a link and link text to an announcement, then save and close it, then reopen it, the Link Button switch is set to off. If i turn it on manually it shows the saved fields just fine though